### PR TITLE
Improves plain renderer exception trace

### PR DESCRIPTION
**Pull request by butschster**
Example of exception trace

```
[Spiral\Views\Exception\RenderException]
Undefined variable $test in vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:23

 1. Spiral\Stempler\ExceptionMapper->mapException() at vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:23
 2. Spiral\Stempler\ExceptionMapper->mapException() at vendor/spiral/framework/src/Bridge/Stempler/src/StemplerView.php:43
 3. Spiral\Stempler\StemplerView->mapException() at vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:34

[ErrorException]
Undefined variable $test in vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:23

 1. Spiral\Exceptions\ExceptionHandler->handleError() at vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:23
 2. __StemplerView__cc8062fe8dd11ddd1bbd36c6e76a2defaf837bde3f2d8db5c225096616eb8484->{closure}() at vendor/spiral/framework/src/Core/src/ContainerScope.php:37
 3. Spiral\Core\ContainerScope::runScope() at vendor/spiral/framework/src/Bridge/Stempler/src/StemplerEngine.php(131) : eval()'d code:31
 4. __StemplerView__cc8062fe8dd11ddd1bbd36c6e76a2defaf837bde3f2d8db5c225096616eb8484->render() at vendor/spiral/framework/src/Views/src/ViewManager.php:147
 5. Spiral\Views\ViewManager->render() at app/src/Api/Web/Controller/HomeController.php:27
 6. App\Api\Web\Controller\HomeController->index()
 7. ReflectionMethod->invokeArgs() at vendor/spiral/framework/src/Hmvc/src/AbstractCore.php:71
 8. Spiral\Core\AbstractCore::Spiral\Core\{closure}() at vendor/spiral/framework/src/Core/src/ContainerScope.php:37
 9. Spiral\Core\ContainerScope::runScope() at vendor/spiral/framework/src/Hmvc/src/AbstractCore.php:72
10. Spiral\Core\AbstractCore->callAction() at vendor/spiral/framework/src/Router/src/CoreHandler.php:96
11. Spiral\Router\CoreHandler->Spiral\Router\{closure}()
12. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
13. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
14. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
15. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
16. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
17. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
18. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Router/src/CoreHandler.php:101
19. Spiral\Router\CoreHandler->Spiral\Router\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
20. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Router/src/CoreHandler.php:104
21. Spiral\Router\CoreHandler->handle() at vendor/spiral/framework/src/Http/src/Pipeline.php:97
22. Spiral\Http\Pipeline::Spiral\Http\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
23. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Http/src/Pipeline.php:98
24. Spiral\Http\Pipeline->handle() at vendor/spiral/framework/src/Http/src/Pipeline.php:55
25. Spiral\Http\Pipeline->process() at vendor/spiral/framework/src/Router/src/Route.php:120
26. Spiral\Router\Route->handle() at vendor/spiral/framework/src/Router/src/Router.php:90
27. Spiral\Router\Router->Spiral\Router\{closure}()
28. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
29. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
30. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
31. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
32. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
33. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
34. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Router/src/Router.php:91
35. Spiral\Router\Router->handle() at vendor/spiral/framework/src/Http/src/Pipeline.php:97
36. Spiral\Http\Pipeline::Spiral\Http\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
37. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Http/src/Pipeline.php:98
38. Spiral\Http\Pipeline->handle() at vendor/spiral/framework/src/Framework/Debug/StateCollector/HttpCollector.php:21
39. Spiral\Debug\StateCollector\HttpCollector->process() at vendor/spiral/framework/src/Http/src/Pipeline.php:72
40. Spiral\Http\Pipeline->Spiral\Http\{closure}()
41. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
42. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
43. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
44. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
45. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
46. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
47. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Http/src/Pipeline.php:89
48. Spiral\Http\Pipeline->handle() at vendor/spiral/framework/src/Http/src/Middleware/JsonPayloadMiddleware.php:39
49. Spiral\Http\Middleware\JsonPayloadMiddleware->process() at vendor/spiral/framework/src/Http/src/Pipeline.php:72
50. Spiral\Http\Pipeline->Spiral\Http\{closure}()
51. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
52. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
53. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
54. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
55. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
56. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
57. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Http/src/Pipeline.php:89
58. Spiral\Http\Pipeline->handle() at vendor/spiral/framework/src/Http/src/Middleware/ErrorHandlerMiddleware.php:43
59. Spiral\Http\Middleware\ErrorHandlerMiddleware->process() at vendor/spiral/framework/src/Http/src/Pipeline.php:72
60. Spiral\Http\Pipeline->Spiral\Http\{closure}()
61. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
62. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
63. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
64. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
65. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
66. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
67. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Http/src/Pipeline.php:89
68. Spiral\Http\Pipeline->handle() at app/src/Application/Middleware/LocaleSelector.php:36
69. App\Application\Middleware\LocaleSelector->process() at vendor/spiral/framework/src/Http/src/Pipeline.php:72
70. Spiral\Http\Pipeline->Spiral\Http\{closure}()
71. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
72. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
73. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
74. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
75. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
76. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
77. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Http/src/Pipeline.php:89
78. Spiral\Http\Pipeline->handle() at vendor/spiral/framework/src/Http/src/Http.php:74
79. Spiral\Http\Http->Spiral\Http\{closure}()
80. ReflectionFunction->invokeArgs() at vendor/spiral/framework/src/Core/src/Internal/Invoker.php:73
81. Spiral\Core\Internal\Invoker->invoke() at vendor/spiral/framework/src/Core/src/Container.php:212
82. Spiral\Core\Container->invoke() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
83. Spiral\Telemetry\AbstractTracer::Spiral\Telemetry\{closure}() at vendor/spiral/framework/src/Core/src/Container.php:161
84. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Telemetry/src/AbstractTracer.php:31
85. Spiral\Telemetry\AbstractTracer->runScope() at vendor/spiral/framework/src/Telemetry/src/NullTracer.php:25
86. Spiral\Telemetry\NullTracer->trace() at vendor/spiral/framework/src/Http/src/Http.php:104
87. Spiral\Http\Http->handle() at vendor/spiral/roadrunner-bridge/src/Http/Dispatcher.php:43
88. Spiral\RoadRunnerBridge\Http\Dispatcher->serve() at vendor/spiral/framework/src/Boot/src/AbstractKernel.php:273
89. Spiral\Boot\AbstractKernel::Spiral\Boot\{closure}() at vendor/spiral/framework/src/Core/src/ContainerScope.php:37
90. Spiral\Core\ContainerScope::runScope() at vendor/spiral/framework/src/Core/src/Container.php:160
91. Spiral\Core\Container->runScope() at vendor/spiral/framework/src/Boot/src/AbstractKernel.php:274
92. Spiral\Boot\AbstractKernel->serve() at app.php:30
```

### CODE REVIEW

**Review by msmakouz — APPROVED** [review]

**Review by roxblnfk — COMMENTED** [review]
Can you cut off path to current workink dir in stacktrace?

File: src/Exceptions/src/Renderer/PlainRenderer.php [hunk 1]
```diff
@@ -42,16 +64,20 @@ public function render(
...
+
+        foreach ($stacktrace as $i => $trace) {
             if (isset($trace['type'], $trace['class'])) {
                 $line = \sprintf(
-                    ' %s%s%s()',
+                    '%s %s%s%s()',
+                    \str_pad(($i + 1) . '.', $pad, ' ', \STR_PAD_LEFT),
                     $trace['class'],
                     $trace['type'],
                     $trace['function']
```
Comment (roxblnfk) [inline src/Exceptions/src/Renderer/PlainRenderer.php:83]: need to add comma

**Review by roxblnfk — CHANGES_REQUESTED** [review]

**Comment by butschster** [comment]
image

**Comment by butschster** [comment]
image

**Comment by butschster** [comment]
image

**Pull request merged**